h2o-core/src/main/java/water/AutoBuffer.java

```diff
@@ -425,14 +426,14 @@ public final int close() {
             assert x == 0xab : "AB.close instead of 0xab sentinel got "+x+", "+this;
             assert _chan != null; // chan set by incoming reader, since we KNOW it is a TCP
             // Write the reader-handshake-byte.
-            ((SocketChannel)_chan).socket().getOutputStream().write(0xcd);
+            SocketChannelUtils.underlyingSocket(_chan).socket().getOutputStream().write(0xcd);
```

Comment:
This is little bit confusion: `underlyingSocket(_chan).socket()`
It sounds that `underlyingSocket` returns a socket already, but it is weird to call `socket()` again